src/ui/components/SettingsPage.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC, useEffect, useState, useCallback } from "react";
+
+const LIVE_COLLECTION = "nimbus-desktop-experiments";
+const PREVIEW_COLLECTION = "nimbus-preview";
+
+const SettingsPage: FC = () => {
+  const [collectionId, setCollectionId] = useState("");
+
+  useEffect(() => {
+    void (async () => {
+      try {
+        const configs = await browser.experiments.nimbus.getCurrentCollection();
+        setCollectionId(configs);
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  }, []);
+
+  const handleToggleChange = useCallback(async () => {
+    if (collectionId === LIVE_COLLECTION) {
+      try {
+        await browser.experiments.nimbus.setCollection(PREVIEW_COLLECTION);
+        setCollectionId(PREVIEW_COLLECTION);
+      } catch (error) {
+        console.error(error);
+      }
+    } else if (collectionId === PREVIEW_COLLECTION) {
+      try {
+        await browser.experiments.nimbus.setCollection(LIVE_COLLECTION);
+        setCollectionId(LIVE_COLLECTION);
+      } catch (error) {
+        console.error(error);
+      }
+    }
```

Comment:
There is no `else` block here, so if the pref has been set to a different collection nothing happens.